mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -1488,8 +1488,6 @@ static FailureOr<OpOperand *> getConsumerFromUses(Value val,
     Operation *consumerOp = opOperand.getOwner();
     if (isa<scf::YieldOp, tensor::ParallelInsertSliceOp>(consumerOp))
       continue;
-    if (operand && *operand != opOperand)
-      return failure();
     // TODO: We have to init result of consumer before scf.for, use
```

Comment:
Wait... We indeed need `if(operand) return failure()` here. Otherwise, the second one(if possible) will overwrite the first one.

Reply:
Yes yes. I was about to push it. Thanks for the catch! PTAL.